lib/utils/utilities.js

```diff
@@ -103,19 +103,23 @@ function getSimulatorModelId (cli, target) {
 function getSimulatorCollection () {
     if (simulatorCollection) return simulatorCollection;
 
-    // Next, figure out the ID of the simulator we found
-    const instrCommand = 'instruments -s devices | grep ^iPhone';
+    /**
+     * We will grab the list of devices and filter only for iPhone items.
+     * xcrun xctrace prints out to stderr so we will redirect the messages to stdout with "2>&1".
+     * Then we will continue to grep as usual.
+     */
+    const cmd = 'xcrun xctrace list devices 2>&1 | grep \'^iPhone\'';
```

Comment:
Yes this is correct, the printout is on stderr.

You can see an example run here, without the redirect, and you will see it on stderr:

https://github.com/erisu/cordova-paramedic/runs/3904672338?check_suite_focus=true#step:9:183

The above link should jump you to line 182, where you will see the content inside stderr.

Reply:
I have just finished fixing iOS 12.x, 13.x, and 14.x.

iOS 12.x and 13.x uses Xcode 11.x and Xcode 11.x does not have `xcrun xctrace`. These two builds continue to use the `instruments` binary. 

Even though the App Store has requirements for which version of Xcode must be used and which iOS version to target, Cordova-iOS 6.x release had supported "Xcode 11 compatibility and minimum version of iOS 11.0" and we were trying to maintain that coverage.

If we were to try and change the process to build everything on Xcode 13 and then run on the various iOS version, I would need to discuss that with a few others for additional feedback.

As for iOS 14.x, it uses Xcode 12.x which introduced the `xcrun xctrace` binary. This works as expected.

There is one caveat: the printout from Xcode 12.x and 13.x are not the same.

Xcode 12.x Printout:
iPhone 12 Pro (14.4) (665718C6-9DA1-457A-BE01-F20E6219B8EF)

Xcode 13.x Printout:
iPhone 12 Pro Simulator (15.0) (50FD9B36-FF9F-4CAE-AEA8-731093C0BFBD)
iPhone 13 Pro Simulator (15.0) (75676197-0A44-43F0-BBF2-42CD231E62B0)

As you can see the word "Simulator" was added in 13.x.

Anyways, in short, the changes I added covers everything except for the Xcode 13.x. 

I think maybe we should merge these changes with your current PR and then merge it into paramedic main branch.

I think we should focus on iOS 15 and Xcode 13.x in one more PR, not this one. Technically these changes will handle using `instruments` if defined for the older version and then defaults to `xcrun` when not defined. This means we prepared the simulator fetching to use proper binary but we need to figure out how to handle the Regular Expression or conditional for all 3 cases.

This is where it was getting tricky. I had tried also with the string replace which you had before, but the tests were still failing because the UDID was not extracted as expected.

Reply:
> Yes this is correct, the printout is on stderr.

That's wild. It doesn't print to stderr for me, but I have found posts from other people online reporting that it is printing to stderr for them as well - it appears to be a bug of some kind. Your redirect works on my setup as well, so I think it's a good solution.

> This is where it was getting tricky. I did try to remove "Simulator" from the string with a string replace, as you had it originaly, but tests continued to fail because it didnt get the simulator's UDID as expected.

How about tweaking the regex so that it looks for the UDID enclosed in brackets `[]` or parens `()`? This regex, combined with a string replace as I was doing before, should work for `instruments`, `xcrun` on Xcode 12, and `xcrun` on Xcode 13.
```
^([a-zA-Z\d ]+) \(([\d.]+)\) [\[\(]([a-zA-Z\d-]*)[\]\)].*
```

I've updated my PR as an example.